### PR TITLE
POST: Send 403 forbidden when a user does not have permission to add …

### PR DESCRIPTION
**Pull request by shriramshankar**
…writers

### CODE REVIEW

File: api/v1/controllers/aspects.js [hunk 1]
```diff
@@ -18,6 +18,8 @@ const doDeleteAllAssoc =
                     require('../helpers/verbs/doDeleteAllBToMAssoc');
 const doDeleteOneAssoc =
                     require('../helpers/verbs/doDeleteOneBToMAssoc');
+const doPostAssoc =
+                    require('../helpers/verbs/doPostBtoMAssoc');
 const doFind = require('../helpers/verbs/doFind');
```
Comment (shriramshankar) [inline api/v1/controllers/aspects.js]: @iamigo @pallavi2209 @annyhe  Any pointers on why Travis is not able to find the file included in line 22? I have checked the file in and the tests pass locally but fail in Travis.

Comment (iamigo) [inline api/v1/controllers/aspects.js]: upper case "T" in BToM the file name but lower case in the require

Comment (shriramshankar) [inline api/v1/controllers/aspects.js]: @iamigo thanks a lot. 😅 Was looking at it for quite some time but was not able to figure it out.

File: api/v1/helpers/verbs/doPostBToMAssoc.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+ * @param {IncomingMessage} req - The request object`
+ * @param {ServerResponse} res - The response object
+ * @param {Function} next - The next middleware function in the stack
+ * @param {Module} props - The module containing the properties of the resource
+ * whose association has to be deleted
+ * @param {String} assocName - The name of the association associated with the
+ * model
+ * @param {Array} assocArray - The instances of a model for which the
+ * association has to be created.
+ *
```
Comment (iamigo) [inline api/v1/helpers/verbs/doPostBToMAssoc.js]: remove extra line here

File: api/v1/helpers/verbs/doPostBToMAssoc.js [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+ */
+function doPostBToMAssoc(req, res, next, props, assocName, assocArray) { // eslint-disable-line
```
Comment (iamigo) [inline api/v1/helpers/verbs/doPostBToMAssoc.js]:
```
function doPostBToMAssoc(req, res, // eslint-disable-line [rule-to-disable]
  next, props, assocName, assocArray) { 
```

**Comment by shriramshankar** [comment]
PR updates with the requested style changes

File: api/v1/helpers/verbs/doDeleteAllBToMAssoc.js [hunk 4]
```diff
@@ -27,7 +27,8 @@ const httpStatus = require('../../constants').httpStatus;
  * model
  *
  */
-function doDeleteAllBToMAssoc(req, res, next, props, assocName) { // eslint-disable-line
+function doDeleteAllBToMAssoc(req, res, next, props, assocName) { // eslint-disable-line max-params
+  console.log('vvv'); // eslint-disable-line no-console
```
Comment (iamigo) [inline api/v1/helpers/verbs/doDeleteAllBToMAssoc.js]: no console.log

File: api/v1/helpers/verbs/doPostBToMAssoc.js [hunk 5]
```diff
@@ -28,9 +28,9 @@ const httpStatus = require('../../constants').httpStatus;
  * model
  * @param {Array} assocArray - The instances of a model for which the
  * association has to be created.
- *
  */
-function doPostBToMAssoc(req, res, next, props, assocName, assocArray) { // eslint-disable-line
+function doPostBToMAssoc(req, res, next, // eslint-disable-line max-params88
```
Comment (iamigo) [inline api/v1/helpers/verbs/doPostBToMAssoc.js]: what's 88?

File: utils/jwtUtil.js [hunk 6]
```diff
@@ -90,12 +90,12 @@ function getUsernameFromToken(req) {
     if (req && req.headers && req.headers.authorization) {
       jwt.verify(req.headers.authorization, env.tokenSecret, {},
       (err, decodedData) => {
-        if (err !== null) {
-          reject(err);
+        if (err !== null || !decodedData) {
+          return reject(err);
```
Comment (annyhe) [inline utils/jwtUtil.js]: @shriramshankar Do we need to return reject or resolve? According to MDN, the reject/resolve returns the corresponding promise.

**Comment by shriramshankar** [comment]
@iamigo @annyhe  PR updated with all the requested changes.

File: tests/api/v1/aspects/postWriters.js [hunk 7]
```diff
@@ -65,7 +66,10 @@ describe('api: aspects: post writers', () => {
     })
     .then((tUsr) => {
       thirdUser = tUsr;
-      userNameArray.push(thirdUser.name);
```
Comment (annyhe) [inline tests/api/v1/aspects/postWriters.js]: @shriramshankar Do we still need to keep the thirdUser variable? Its value doesn't seem to be used.

File: utils/jwtUtil.js (hunk 6 above)
Comment (shriramshankar) [inline utils/jwtUtil.js]: The "return" needed to be added here. The resolve and reject was not returning the promise.

**Pull request merged**